provisioning_client/devdoc/azure_provisioning_faq.md

```diff
@@ -0,0 +1,39 @@
+# Azure Iot Provisioning Client FAQ
+
+Here is where we shall try to answer question pertaining to the IoT Provisioning Client.  Please read the following documents for an overview of the DPS service.
+
+## x509 Provisioning
+
+- How does Provisining match the x509 certificates to an enrollment?
+
+  - The Provisioning Client add the x509 certificate (more on the certificate in a second) on the TLS stream.  The DPS service uses this certificate to look up the proper enrollment record.
+
+- Where does the x509 certificate comes from?
+
+  - This question depends on the situation:
+
+    - **Development**: The SDK ships with a development repo that generates test x509 with a canned key.  This enables the developer to quickly get up and running to test their solutions.
+
+    - **Production**: For productions situations the developer should create a custom HSM library to retrieve the certificate from a hardware backed HSM or a software solution (for more information on this please see [using custom hsm](https://github.com/Azure/azure-iot-sdk-c/blob/master/provisioning_client/devdoc/using_custom_hsm.md))
+
+- What if I want the certificate private key to never leave the hardware?
+
+  - One of the benifits of having an HSM is that it can ensure that the private key will never leave the hardware device, but to get this functionality it will require a little more code.
+
+    - You will need to write a custom HSM to be able to extract the certificate and return the alias private key to the SDK (more on this in a moment) see the [custom hsm sample](https://github.com/Azure/azure-iot-sdk-c/blob/master/provisioning_client/samples/custom_hsm_example/custom_hsm_example.c) in the SDK.
+
+    - You will need to have a tls engine that can communicate with the target hardware that is connected to the device.
```

Comment:
s/tls/TLS

Are there particular TLS engines we have tried and feel comfortable mentioning?